test/document_stores/test_pinecone.py

```diff
@@ -417,3 +417,15 @@ def test_multilayer_dict(self, doc_store_with_docs: PineconeDocumentStore):
 
         assert len(retrieved_docs) == 1
         assert retrieved_docs[0].meta == multilayer_meta
+
+    @pytest.mark.unit
+    def test_skip_validating_empty_embeddings(self, ds: PineconeDocumentStore):
+        document = Document(id="0", content="test")
+        retriever = MockBaseRetriever(document_store=ds, mock_document=document)
+        ds.write_documents(documents=[document])
+        ds._validate_embeddings_shape = MagicMock()
+
+        ds.update_embeddings(retriever)
+        ds._validate_embeddings_shape.assert_called_once()
+        ds.update_embeddings(retriever, update_existing_embeddings=False)
+        ds._validate_embeddings_shape.assert_called_once()
```

Comment:
Not sure we are testing lines 508-513? We would need to have `embedding.size == 0` but the mocked retriever always returns something I think

Reply:
Yes, it does. We get `embedding.size==0` because `embeddings = retriever.embed_documents(document_batch)` results in no embeddings even for the mocked retriever if `document_batch` is an empty list. That's the case if we don't need to update the embedding of any document.